### PR TITLE
Address PR feedback from #2329

### PR DESCRIPTION
**Pull request by cshung**
@GrabYourPitchforks
@ahsonkhan

### CODE REVIEW

File: src/System.Text.JsonLab/System/Text/Json/JsonWriterUtf16.cs [hunk 1]
```diff
@@ -570,10 +570,17 @@ public void WriteNull()
...
-            _bufferWriter.Advance(bytesNeeded);
+            if (BitConverter.IsLittleEndian)
+            {
+                Debug.Assert(byteBuffer.Slice(idx * 2).Length >= JsonConstants.NullValueUtf16LE.Length);
+                JsonConstants.NullValueUtf16LE.CopyTo(byteBuffer.Slice(idx * 2));
+                _bufferWriter.Advance(bytesNeeded);
+            }
+            else
+            {
+                _bufferWriter.Advance(bytesNeeded);
```
Comment (ahsonkhan) [inline src/System.Text.JsonLab/System/Text/Json/JsonWriterUtf16.cs]: Add a NullValueUtf16BE and use that instead?

Alternatively, try to isolate and measure the difference in performance between calling the UTF-16 to UTF-8 encode method for a small input (like 4-byte null array) and compare that to directly copying a pre-computed byte array. If there is little performance impact, we could keep the encoding step, which appropriately deals with the endianness.

**Comment by ahsonkhan** [comment]
@dotnet-bot test this please

@GrabYourPitchforks, any feedback, especially on the other points you mentioned previously?
- [ensure buffer](https://github.com/dotnet/corefxlab/pull/2329#discussion_r192606825)
- [integer overflow][(https://github.com/dotnet/corefxlab/pull/2329#discussion_r192606868)

**Comment by cshung** [comment]
@ahsonkhan 
I have merged the conflicts. Shall we check this in so that we fix the issue on big-endian machine? We can keep the issue open to track the alignment and overflow if necessary.

**Comment by ahsonkhan** [comment]
> I have merged the conflicts. Shall we check this in so that we fix the issue on big-endian machine? We can keep the issue open to track the alignment and overflow if necessary.

Yes, sounds good to me.

**Pull request merged**